build error in spack

I'm getting the following error when building in spack:

```
1 error found in build log:
     81    -- Looking for pthread_create in pthreads
     82    -- Looking for pthread_create in pthreads - not found
     83    -- Looking for pthread_create in pthread
     84    -- Looking for pthread_create in pthread - found
     85    -- Found Threads: TRUE
     86    -- hip::amdhip64 is SHARED_LIBRARY
  >> 87    CMake Error at cmake/Packages.cmake:120 (find_package):
     88      Could not find a package configuration file provided by "ROCmVersi
           on" with
     89      any of the following names:
     90    
     91        ROCmVersionConfig.cmake
     92        rocmversion-config.cmake
     93    

```

If it helps, here is my spec:

```
[lee218@rzvernal11:spack]$ ./bin/spack spec omnitrace@main %gcc@8.3.1
==> Warning: Missing a source id for omnitrace@main
Input spec
--------------------------------
omnitrace@main%gcc@8.3.1

Concretized
--------------------------------
omnitrace@main%gcc@8.3.1~caliper~ipo~mpi+mpi_headers+ompt+papi~perfetto_tools~python+rocm~strip~tau build_type=Release arch=cray-rhel8-zen
    ^cmake@3.23.1%gcc@8.3.1~doc+ncurses+ownlibs~qt build_type=Release arch=cray-rhel8-zen
    ^dyninst@12.2.0%gcc@8.3.1~ipo+openmp~stat_dysect~static build_type=RelWithDebInfo arch=cray-rhel8-zen
        ^boost@1.79.0%gcc@8.3.1+atomic+chrono~clanglibcpp~container~context~contract~coroutine+date_time~debug~exception~fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace+system~taggedlayout~test+thread+timer~type_erasure~versionedlayout~wave cxxstd=98 patches=57a8401,a440f96 visibility=hidden arch=cray-rhel8-zen
        ^elfutils@0.186%gcc@8.3.1~bzip2~debuginfod+nls~xz arch=cray-rhel8-zen
            ^gettext@0.21%gcc@8.3.1+bzip2+curses+git~libunistring+libxml2+tar+xz arch=cray-rhel8-zen
                ^bzip2@1.0.8%gcc@8.3.1~debug~pic+shared arch=cray-rhel8-zen
                    ^diffutils@3.6%gcc@8.3.1 arch=cray-rhel8-zen
                ^libiconv@1.16%gcc@8.3.1 libs=shared,static arch=cray-rhel8-zen
                ^libxml2@2.10.1%gcc@8.3.1~python arch=cray-rhel8-zen
                    ^pkgconf@1.4.2%gcc@8.3.1 arch=cray-rhel8-zen
                    ^xz@5.2.5%gcc@8.3.1~pic libs=shared,static arch=cray-rhel8-zen
                    ^zlib@1.2.12%gcc@8.3.1+optimize+pic+shared patches=0d38234 arch=cray-rhel8-zen
                ^ncurses@6.3%gcc@8.3.1~symlinks+termlib abi=none arch=cray-rhel8-zen
                ^tar@1.30%gcc@8.3.1 zip=pigz arch=cray-rhel8-zen
            ^m4@1.4.18%gcc@8.3.1+sigsegv patches=3877ab5,fc9b616 arch=cray-rhel8-zen
        ^intel-tbb@2020.3%gcc@8.3.1~ipo+shared+tm build_type=RelWithDebInfo cxxstd=default patches=62ba015,ce1fb16,d62cb66 arch=cray-rhel8-zen
        ^libiberty@2.37%gcc@8.3.1+pic arch=cray-rhel8-zen
    ^hip@5.2.1%gcc@8.3.1~ipo build_type=Release patches=7ed1232 arch=cray-rhel8-zen
        ^comgr@5.2.1%gcc@8.3.1~ipo build_type=Release arch=cray-rhel8-zen
            ^llvm-amdgpu@5.2.1%gcc@8.3.1~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=a08bbe1 arch=cray-rhel8-zen
                ^python@3.9.13%gcc@8.3.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,ebdca64,f2fd060 arch=cray-rhel8-zen
                    ^expat@2.4.8%gcc@8.3.1+libbsd arch=cray-rhel8-zen
                        ^libbsd@0.11.5%gcc@8.3.1 arch=cray-rhel8-zen
                            ^libmd@1.0.4%gcc@8.3.1 arch=cray-rhel8-zen
                    ^gdbm@1.19%gcc@8.3.1 arch=cray-rhel8-zen
                        ^readline@8.1.2%gcc@8.3.1 arch=cray-rhel8-zen
                    ^libffi@3.4.2%gcc@8.3.1 arch=cray-rhel8-zen
                    ^openssl@1.0.2k-fips%gcc@8.3.1~docs~shared certs=mozilla arch=cray-rhel8-zen
                    ^sqlite@3.39.2%gcc@8.3.1+column_metadata+dynamic_extensions+fts~functions+rtree arch=cray-rhel8-zen
                    ^util-linux-uuid@2.37.4%gcc@8.3.1 arch=cray-rhel8-zen
                ^z3@4.8.16%gcc@8.3.1~gmp~ipo~python build_type=RelWithDebInfo arch=cray-rhel8-zen
            ^rocm-cmake@5.2.1%gcc@8.3.1~ipo build_type=Release arch=cray-rhel8-zen
        ^glx@1.4%gcc@8.3.1 arch=cray-rhel8-zen
            ^mesa@22.1.2%gcc@8.3.1+glx+llvm+opengl~opengles+osmesa~strip buildtype=release default_library=shared patches=ee737d1 arch=cray-rhel8-zen
                ^bison@3.0.4%gcc@8.3.1 patches=b72914f arch=cray-rhel8-zen
                ^flex@2.6.1%gcc@8.3.1+lex~nls arch=cray-rhel8-zen
                ^glproto@1.4.17%gcc@8.3.1 arch=cray-rhel8-zen
                    ^util-macros@1.19.3%gcc@8.3.1 arch=cray-rhel8-zen
                ^libunwind@1.6.2%gcc@8.3.1~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib components=none libs=shared,static arch=cray-rhel8-zen
                ^libx11@1.7.0%gcc@8.3.1 arch=cray-rhel8-zen
                    ^inputproto@2.3.2%gcc@8.3.1 arch=cray-rhel8-zen
                    ^kbproto@1.0.7%gcc@8.3.1 arch=cray-rhel8-zen
                    ^libxcb@1.14%gcc@8.3.1 arch=cray-rhel8-zen
                        ^libpthread-stubs@0.4%gcc@8.3.1 arch=cray-rhel8-zen
                        ^libxau@1.0.8%gcc@8.3.1 arch=cray-rhel8-zen
                            ^xproto@7.0.31%gcc@8.3.1 arch=cray-rhel8-zen
                        ^libxdmcp@1.1.2%gcc@8.3.1 arch=cray-rhel8-zen
                        ^xcb-proto@1.14.1%gcc@8.3.1 arch=cray-rhel8-zen
                    ^perl@5.34.1%gcc@8.3.1+cpanm+shared+threads arch=cray-rhel8-zen
                        ^berkeley-db@18.1.40%gcc@8.3.1+cxx~docs+stl patches=26090f4,b231fcc arch=cray-rhel8-zen
                    ^xextproto@7.3.0%gcc@8.3.1 arch=cray-rhel8-zen
                    ^xtrans@1.3.5%gcc@8.3.1 arch=cray-rhel8-zen
                ^libxext@1.3.3%gcc@8.3.1 arch=cray-rhel8-zen
                ^libxt@1.1.5%gcc@8.3.1 arch=cray-rhel8-zen
                    ^libice@1.0.9%gcc@8.3.1 arch=cray-rhel8-zen
                    ^libsm@1.2.3%gcc@8.3.1 arch=cray-rhel8-zen
                ^meson@0.63.2%gcc@8.3.1 patches=9c87472,aa6c50d arch=cray-rhel8-zen
                    ^ninja@1.11.0%gcc@8.3.1 arch=cray-rhel8-zen
                    ^py-pip@22.1.2%gcc@8.3.1 arch=cray-rhel8-zen
                    ^py-setuptools@65.0.0%gcc@8.3.1 arch=cray-rhel8-zen
                    ^py-wheel@0.37.1%gcc@8.3.1 arch=cray-rhel8-zen
                ^py-mako@1.1.6%gcc@8.3.1 arch=cray-rhel8-zen
                    ^py-markupsafe@2.0.1%gcc@8.3.1 arch=cray-rhel8-zen
                ^xrandr@1.5.0%gcc@8.3.1 arch=cray-rhel8-zen
                    ^libxrandr@1.5.0%gcc@8.3.1 arch=cray-rhel8-zen
                        ^libxrender@0.9.10%gcc@8.3.1 arch=cray-rhel8-zen
                            ^renderproto@0.11.1%gcc@8.3.1 arch=cray-rhel8-zen
                        ^randrproto@1.5.0%gcc@8.3.1 arch=cray-rhel8-zen
        ^hsa-rocr-dev@5.2.1%gcc@8.3.1+image~ipo+shared build_type=Release patches=71e6851 arch=cray-rhel8-zen
            ^hsakmt-roct@5.2.1%gcc@8.3.1~ipo+shared build_type=Release patches=f926273 arch=cray-rhel8-zen
                ^libdrm@2.4.110%gcc@8.3.1~docs arch=cray-rhel8-zen
                    ^libpciaccess@0.16%gcc@8.3.1 arch=cray-rhel8-zen
                        ^libtool@2.4.6%gcc@8.3.1 arch=cray-rhel8-zen
                ^numactl@2.0.14%gcc@8.3.1 patches=4e1d78c,62fc8a8,ff37630 arch=cray-rhel8-zen
                    ^autoconf@2.69%gcc@8.3.1 patches=7793209 arch=cray-rhel8-zen
                    ^automake@1.16.1%gcc@8.3.1 arch=cray-rhel8-zen
            ^xxd-standalone@8.2.1201%gcc@8.3.1 arch=cray-rhel8-zen
        ^perl-file-which@1.22%gcc@8.3.1 arch=cray-rhel8-zen
        ^perl-uri-encode@1.1.1%gcc@8.3.1 arch=cray-rhel8-zen
            ^perl-module-build@0.4224%gcc@8.3.1 arch=cray-rhel8-zen
        ^rocminfo@5.2.1%gcc@8.3.1~ipo build_type=Release arch=cray-rhel8-zen
        ^roctracer-dev-api@5.2.1%gcc@8.3.1 arch=cray-rhel8-zen
    ^papi@6.0.0.1%gcc@8.3.1~cuda+example~infiniband~lmsensors~nvml~powercap~rapl~rocm~rocm_smi~sde+shared~static_tools arch=cray-rhel8-zen
    ^rocm-smi-lib@5.2.1%gcc@8.3.1~ipo+shared build_type=Release patches=8bc40cc arch=cray-rhel8-zen
    ^rocprofiler-dev@5.2.1%gcc@8.3.1~ipo build_type=Release patches=16754a1 arch=cray-rhel8-zen
    ^roctracer-dev@5.2.1%gcc@8.3.1~ipo build_type=Release arch=cray-rhel8-zen
        ^py-cppheaderparser@2.7.4%gcc@8.3.1 arch=cray-rhel8-zen
            ^py-ply@3.11%gcc@8.3.1 arch=cray-rhel8-zen
```

--- a/cmake/Modules/FindROCmVersion.cmake
+++ b/cmake/Modules/FindROCmVersion.cmake
@@ -32,161 +32,169 @@ These variables are relevant for the find procedure::
   ROCmVersion_DIR               - Root location for <ROCM_PATH>
 #]=======================================================================]
 
+set(ROCmVersion_VARIABLES
+    EPOCH
+    MAJOR
+    MINOR
+    PATCH
+    TWEAK
+    REVISION
+    TRIPLE
+    NUMERIC
+    CANONICAL
+    FULL)
+
+function(ROCM_VERSION_MESSAGE _TYPE)
+    if(ROCmVersion_DEBUG)
+        message(${_TYPE} "[ROCmVersion] ${ARGN}")
+    endif()
+endfunction()
+
+# read a .info/version* file and propagate the variables to the calling scope
+function(ROCM_VERSION_COMPUTE FULL_VERSION_STRING _VAR_PREFIX)
+
+    # remove any line endings
+    string(REGEX REPLACE "(\n|\r)" "" FULL_VERSION_STRING "${FULL_VERSION_STRING}")
+
+    # store the full version so it can be set later
+    set(FULL_VERSION "${FULL_VERSION_STRING}")
+
+    # get number and remove from full version string
+    string(REGEX REPLACE "([0-9]+)\:(.*)" "\\1" EPOCH_VERSION "${FULL_VERSION_STRING}")
+    string(REGEX REPLACE "([0-9]+)\:(.*)" "\\2" FULL_VERSION_STRING
+                         "${FULL_VERSION_STRING}")
+
+    if(EPOCH_VERSION STREQUAL FULL_VERSION)
+        set(EPOCH_VERSION)
+    endif()
+
+    # get number and remove from full version string
+    string(REGEX REPLACE "([0-9]+)(.*)" "\\1" MAJOR_VERSION "${FULL_VERSION_STRING}")
+    string(REGEX REPLACE "([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
+                         "${FULL_VERSION_STRING}")
+
+    # get number and remove from full version string
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" MINOR_VERSION "${FULL_VERSION_STRING}")
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
+                         "${FULL_VERSION_STRING}")
+
+    # get number and remove from full version string
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" PATCH_VERSION "${FULL_VERSION_STRING}")
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
+                         "${FULL_VERSION_STRING}")
+
+    if(NOT PATCH_VERSION LESS 100)
+        set(PATCH_VERSION 0)
+    endif()
+
+    # get number and remove from full version string
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" TWEAK_VERSION "${FULL_VERSION_STRING}")
+    string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
+                         "${FULL_VERSION_STRING}")
+
+    # get number
+    string(REGEX REPLACE "-([0-9A-Za-z+~]+)" "\\1" REVISION_VERSION
+                         "${FULL_VERSION_STRING}")
+
+    set(CANONICAL_VERSION)
+    set(_MAJOR_SEP ":")
+    set(_MINOR_SEP ".")
+    set(_PATCH_SEP ".")
+    set(_TWEAK_SEP ".")
+    set(_REVISION_SEP "-")
+
+    foreach(_V EPOCH MAJOR MINOR PATCH TWEAK REVISION)
+        if(${_V}_VERSION)
+            set(CANONICAL_VERSION "${CANONICAL_VERSION}${_${_V}_SEP}${${_V}_VERSION}")
+        else()
+            set(CANONICAL_VERSION "${CANONICAL_VERSION}${_${_V}_SEP}0")
+        endif()
+    endforeach()
+
+    set(_MAJOR_SEP "")
+
+    foreach(_V MAJOR MINOR PATCH)
+        if(${_V}_VERSION)
+            set(TRIPLE_VERSION "${TRIPLE_VERSION}${_${_V}_SEP}${${_V}_VERSION}")
+        else()
+            set(TRIPLE_VERSION "${TRIPLE_VERSION}${_${_V}_SEP}0")
+        endif()
+    endforeach()
+
+    math(
+        EXPR
+        NUMERIC_VERSION
+        "(10000 * (${MAJOR_VERSION}+0)) + (100 * (${MINOR_VERSION}+0)) + (${PATCH_VERSION}+0)"
+        )
+
+    # propagate to parent scopes
+    foreach(_V ${ROCmVersion_VARIABLES})
+        set(${_VAR_PREFIX}_${_V}_VERSION
+            ${${_V}_VERSION}
+            PARENT_SCOPE)
+    endforeach()
+endfunction()
+
+# this macro watches for changes in the variables and unsets the remaining cache varaible
+# when they change
+function(ROCM_VERSION_WATCH_FOR_CHANGE _var)
+    set(_rocm_version_watch_var_name ROCmVersion_WATCH_VALUE_${_var})
+
+    if(DEFINED ${_rocm_version_watch_var_name})
+        if("${${_var}}" STREQUAL "${${_rocm_version_watch_var_name}}")
+            if(NOT "${${_var}}" STREQUAL "")
+                rocm_version_message(STATUS "${_var} :: ${${_var}}")
+            endif()
+
+            list(REMOVE_ITEM _REMAIN_VARIABLES ${_var})
+            set(_REMAIN_VARIABLES
+                "${_REMAIN_VARIABLES}"
+                PARENT_SCOPE)
+            return()
+        else()
+            rocm_version_message(
+                STATUS
+                "${_var} changed :: ${${_rocm_version_watch_var_name}} --> ${${_var}}")
+
+            foreach(_V ${_REMAIN_VARIABLES})
+                rocm_version_message(
+                    STATUS "${_var} changed :: Unsetting cache variable ${_V}...")
+                unset(${_V} CACHE)
+            endforeach()
+        endif()
+    else()
+        if(NOT "${${_var}}" STREQUAL "")
+            rocm_version_message(STATUS "${_var} :: ${${_var}}")
+        endif()
+    endif()
+
+    # store the value for the next run
+    set(${_rocm_version_watch_var_name}
+        "${${_var}}"
+        CACHE INTERNAL "Last value of ${_var}" FORCE)
+endfunction()
+
 # scope this to a function to avoid leaking local variables
 function(ROCM_VERSION_PARSE_VERSION_FILES)
-
-    function(ROCM_VERSION_MESSAGE _TYPE)
-        if(ROCmVersion_DEBUG)
-            message(${_TYPE} "[ROCmVersion] ${ARGN}")
-        endif()
-    endfunction()
 
     # the list of variables set by module. when one of these changes, we need to unset the
     # cache variables after it
     set(_ALL_VARIABLES)
-    foreach(
-        _V
-        EPOCH
-        MAJOR
-        MINOR
-        PATCH
-        TWEAK
-        REVISION
-        TRIPLE
-        NUMERIC
-        FULL
-        CANONICAL)
+
+    foreach(_V ${ROCmVersion_VARIABLES})
         list(APPEND _ALL_VARIABLES ROCmVersion_${_V}_VERSION)
     endforeach()
     set(_REMAIN_VARIABLES ${_ALL_VARIABLES})
 
-    # this macro watches for changes in the variables and unsets the remaining cache
-    # varaible when they change
-    function(ROCM_VERSION_WATCH_FOR_CHANGE _var)
-        set(_rocm_version_watch_var_name ROCmVersion_WATCH_VALUE_${_var})
-        if(DEFINED ${_rocm_version_watch_var_name})
-            if("${${_var}}" STREQUAL "${${_rocm_version_watch_var_name}}")
-                if(NOT "${${_var}}" STREQUAL "")
-                    rocm_version_message(STATUS "${_var} :: ${${_var}}")
-                endif()
-                list(REMOVE_ITEM _REMAIN_VARIABLES ${_var})
-                set(_REMAIN_VARIABLES
-                    "${_REMAIN_VARIABLES}"
-                    PARENT_SCOPE)
-                return()
-            else()
-                rocm_version_message(
-                    STATUS
-                    "${_var} changed :: ${${_rocm_version_watch_var_name}} --> ${${_var}}"
-                    )
-                foreach(_V ${_REMAIN_VARIABLES})
-                    rocm_version_message(
-                        STATUS "${_var} changed :: Unsetting cache variable ${_V}...")
-                    unset(${_V} CACHE)
-                endforeach()
-            endif()
-        else()
-            if(NOT "${${_var}}" STREQUAL "")
-                rocm_version_message(STATUS "${_var} :: ${${_var}}")
-            endif()
-        endif()
-
-        # store the value for the next run
-        set(${_rocm_version_watch_var_name}
-            "${${_var}}"
-            CACHE INTERNAL "Last value of ${_var}" FORCE)
-    endfunction()
-
     # read a .info/version* file and propagate the variables to the calling scope
     function(ROCM_VERSION_READ_FILE _FILE _VAR_PREFIX)
         file(READ "${_FILE}" FULL_VERSION_STRING LIMIT_COUNT 1)
-
-        # remove any line endings
-        string(REGEX REPLACE "(\n|\r)" "" FULL_VERSION_STRING "${FULL_VERSION_STRING}")
-
-        # store the full version so it can be set later
-        set(FULL_VERSION "${FULL_VERSION_STRING}")
-
-        # get number and remove from full version string
-        string(REGEX REPLACE "([0-9]+)\:(.*)" "\\1" EPOCH_VERSION
-                             "${FULL_VERSION_STRING}")
-        string(REGEX REPLACE "([0-9]+)\:(.*)" "\\2" FULL_VERSION_STRING
-                             "${FULL_VERSION_STRING}")
-
-        if(EPOCH_VERSION STREQUAL FULL_VERSION)
-            set(EPOCH_VERSION)
-        endif()
-
-        # get number and remove from full version string
-        string(REGEX REPLACE "([0-9]+)(.*)" "\\1" MAJOR_VERSION "${FULL_VERSION_STRING}")
-        string(REGEX REPLACE "([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
-                             "${FULL_VERSION_STRING}")
-
-        # get number and remove from full version string
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" MINOR_VERSION
-                             "${FULL_VERSION_STRING}")
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
-                             "${FULL_VERSION_STRING}")
-
-        # get number and remove from full version string
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" PATCH_VERSION
-                             "${FULL_VERSION_STRING}")
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
-                             "${FULL_VERSION_STRING}")
-
-        # get number and remove from full version string
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\1" TWEAK_VERSION
-                             "${FULL_VERSION_STRING}")
-        string(REGEX REPLACE "\.([0-9]+)(.*)" "\\2" FULL_VERSION_STRING
-                             "${FULL_VERSION_STRING}")
-
-        # get number
-        string(REGEX REPLACE "-([0-9A-Za-z+~]+)" "\\1" REVISION_VERSION
-                             "${FULL_VERSION_STRING}")
-
-        set(CANONICAL_VERSION)
-        set(_MAJOR_SEP ":")
-        set(_MINOR_SEP ".")
-        set(_PATCH_SEP ".")
-        set(_TWEAK_SEP ".")
-        set(_REVISION_SEP "-")
-        foreach(_V EPOCH MAJOR MINOR PATCH TWEAK REVISION)
-            if(${_V}_VERSION)
-                set(CANONICAL_VERSION "${CANONICAL_VERSION}${_${_V}_SEP}${${_V}_VERSION}")
-            else()
-                set(CANONICAL_VERSION "${CANONICAL_VERSION}${_${_V}_SEP}0")
-            endif()
-        endforeach()
-        set(_MAJOR_SEP "")
-        foreach(_V MAJOR MINOR PATCH)
-            if(${_V}_VERSION)
-                set(TRIPLE_VERSION "${TRIPLE_VERSION}${_${_V}_SEP}${${_V}_VERSION}")
-            else()
-                set(TRIPLE_VERSION "${TRIPLE_VERSION}${_${_V}_SEP}0")
-            endif()
-        endforeach()
-
-        math(
-            EXPR
-            NUMERIC_VERSION
-            "(10000 * (${MAJOR_VERSION}+0)) + (100 * (${MINOR_VERSION}+0)) + (${PATCH_VERSION}+0)"
-            )
+        rocm_version_compute("${FULL_VERSION_STRING}" "${_VAR_PREFIX}")
 
         # propagate to parent scopes
-        foreach(
-            _V
-            EPOCH
-            MAJOR
-            MINOR
-            PATCH
-            TWEAK
-            REVISION
-            TRIPLE
-            NUMERIC
-            CANONICAL
-            FULL)
+        foreach(_V ${ROCmVersion_VARIABLES})
             set(${_VAR_PREFIX}_${_V}_VERSION
-                ${${_V}_VERSION}
+                ${${_VAR_PREFIX}_${_V}_VERSION}
                 PARENT_SCOPE)
         endforeach()
     endfunction()
@@ -269,18 +277,8 @@ function(ROCM_VERSION_PARSE_VERSION_FILES)
         string(REPLACE "." "_" _B "${_B}")
         string(REPLACE "-" "_" _B "${_B}")
         rocm_version_read_file(${_F} ${_B})
-        foreach(
-            _V
-            EPOCH
-            MAJOR
-            MINOR
-            PATCH
-            TWEAK
-            REVISION
-            TRIPLE
-            NUMERIC
-            FULL
-            CANONICAL)
+
+        foreach(_V ${ROCmVersion_VARIABLES})
             set(_CACHE_VAR ROCmVersion_${_V}_VERSION)
             set(_LOCAL_VAR ${_B}_${_V}_VERSION)
             set(ROCmVersion_${_V}_VERSION

--- a/cmake/Packages.cmake
+++ b/cmake/Packages.cmake
@@ -117,10 +117,27 @@ if(OMNITRACE_USE_HIP
     if(NOT ROCmVersion_FOUND)
         find_package(hip ${omnitrace_FIND_QUIETLY} REQUIRED HINTS
                      ${OMNITRACE_DEFAULT_ROCM_PATH} PATHS ${OMNITRACE_DEFAULT_ROCM_PATH})
-        find_package(ROCmVersion REQUIRED HINTS ${ROCM_PATH} PATHS ${ROCM_PATH})
+        if(SPACK_BUILD)
+            find_package(ROCmVersion HINTS ${ROCM_PATH} PATHS ${ROCM_PATH})
+        else()
+            find_package(ROCmVersion REQUIRED HINTS ${ROCM_PATH} PATHS ${ROCM_PATH})
+        endif()
     endif()
 
-    list(APPEND CMAKE_PREFIX_PATH ${ROCmVersion_DIR})
+    if(NOT ROCmVersion_FOUND)
+        rocm_version_compute("${hip_VERSION}" _local)
+
+        foreach(_V ${ROCmVersion_VARIABLES})
+            set(_CACHE_VAR ROCmVersion_${_V}_VERSION)
+            set(_LOCAL_VAR _local_${_V}_VERSION)
+            set(ROCmVersion_${_V}_VERSION
+                "${${_LOCAL_VAR}}"
+                CACHE STRING "ROCm ${_V} version")
+            rocm_version_watch_for_change(${_CACHE_VAR})
+        endforeach()
+    else()
+        list(APPEND CMAKE_PREFIX_PATH ${ROCmVersion_DIR})
+    endif()
 
     set(OMNITRACE_ROCM_VERSION ${ROCmVersion_FULL_VERSION})
     set(OMNITRACE_HIP_VERSION_MAJOR ${ROCmVersion_MAJOR_VERSION})
@@ -884,6 +901,10 @@ set(CMAKE_INSTALL_PYTHONDIR ${OMNITRACE_INSTALL_PYTHONDIR})
 # Compile definitions
 #
 # ----------------------------------------------------------------------------------------#
+
+if("${CMAKE_BUILD_TYPE}" MATCHES "Release" AND NOT OMNITRACE_BUILD_DEBUG)
+    add_target_flag_if_avail(omnitrace-compile-options "-g1")
+endif()
 
 foreach(_LIB ${OMNITRACE_EXTENSION_LIBRARIES})
     get_target_property(_COMPILE_DEFS ${_LIB} INTERFACE_COMPILE_DEFINITIONS)

